### PR TITLE
Vuln cache, pmap usage, and cache on executionHandler mainline

### PR DESCRIPTION
**Pull request by kenanwarren**
The current endpoints used to get vulnerability details and asset information are heavily rate limited consequently causing job execution timeouts. To resolve this issue we've migrated to the bulk export apis to reduce pressure on the tenable api.

### CODE REVIEW

File: src/tenable/index.test.ts [hunk 1]
```diff
@@ -90,3 +90,137 @@ describe("TenableAssetExportCache", () => {
     nock.restore();
   });
 });
+
```
Comment (mknoedel) [inline src/tenable/index.test.ts:93]: Nice tests!

File: src/tenable/createAssetExportCache.ts [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    ({
+      status,
+      chunks_available: chunksAvailable,
+    } = await client.fetchAssetsExportStatus(exportUuid));
+  }
+
+  const chunkResponses = await pMap(
+    chunksAvailable,
+    async chunkId => await client.fetchAssetsExportChunk(exportUuid, chunkId),
+    { concurrency: 3 },
```
Comment (mknoedel) [inline src/tenable/createAssetExportCache.ts:45]: Is there any logic behind this concurrency?

File: src/tenable/createAssetExportCache.ts [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+  let {
+    status,
+    chunks_available: chunksAvailable,
+  } = await client.fetchAssetsExportStatus(exportUuid);
+
+  while ([ExportStatus.Processing, ExportStatus.Queued].includes(status)) {
+    ({
+      status,
+      chunks_available: chunksAvailable,
+    } = await client.fetchAssetsExportStatus(exportUuid));
```
Comment (mknoedel) [inline src/tenable/createAssetExportCache.ts:39]: Are we concerned about getting stuck in a processing loop?

File: src/tenable/createVulnerabilityExportCache.ts [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+  };
+  const { export_uuid: exportUuid } = await client.exportVulnerabilities(
+    options,
+  );
+  let {
+    status,
+    chunks_available: chunksAvailable,
+  } = await client.fetchVulnerabilitiesExportStatus(exportUuid);
+
+  while ([ExportStatus.Processing, ExportStatus.Queued].includes(status)) {
```
Comment (mknoedel) [inline src/tenable/createVulnerabilityExportCache.ts:88]: same questions as above for this pattern.

**Review by mknoedel — APPROVED** [review]
Looks great! It would be nice to add some description in the PR for me to read the purpose of you PR as I'm not too sure why some code is being replaced.

File: src/tenable/createAssetExportCache.ts (hunk 2 above)
Comment (kenanwarren) [reply src/tenable/createAssetExportCache.ts:45]: On a lot of the other endpoints they have pretty aggressive rate limiting which caused the initial issue. I've never actually encountered a rate limit on these endpoints, but my data sample is a lot lower than what our customers might have.

File: src/tenable/createAssetExportCache.ts (hunk 3 above)
Comment (kenanwarren) [reply src/tenable/createAssetExportCache.ts:39]: That's a good point, I'll cook something up in a follow up PR.

**Pull request merged**